src/detect-mqtt-connect-flags.c

```diff
@@ -134,30 +134,36 @@ static int DetectMQTTConnectFlagsMatch(DetectEngineThreadCtx *det_ctx,
 static DetectMQTTConnectFlagsData *DetectMQTTConnectFlagsParse(const char *rawstr)
 {
     DetectMQTTConnectFlagsData *de = NULL;
+    char *copy = NULL;
     int ret = 0;
     int ov[MAX_SUBSTRINGS];
 
     ret = DetectParsePcreExec(&parse_regex, rawstr, 0, 0, ov, MAX_SUBSTRINGS);
     if (ret < 1) {
         SCLogError(SC_ERR_PCRE_MATCH, "invalid flag definition: %s", rawstr);
-        return NULL;
+        goto error;
     }
 
     de = SCCalloc(1, sizeof(DetectMQTTConnectFlagsData));
     if (unlikely(de == NULL))
-        return NULL;
+        goto error;
```

Comment:
returning NULL here was fine. I think the issue coverity flagged is that after this we don't have to check `de != NULL` anymore

Reply:
True. That check was requested in a previous review though -- I would be happy to remove it in these 3 cases as I agree it it only complicates things.